firmware/screen/tools/long_cmd_fix_post.py

```
@@ -40,12 +44,40 @@
     arq = f'"{ar}"'
 
+    # CC / CXX / LINK keep TEMPFILE wrapping because gcc/g++ honour @response-file.
+    # AR does NOT — passing @file to xtensa ar.exe fails ("invalid option -- @"
+    # because binutils 2.31.1 in this toolchain predates @file support). And the
+    # full LVGL archive command is ~10K chars which exceeds Windows' 8191 cmd.exe
+    # shell limit. Solution: invoke ar in batches through a Python action that
+    # uses subprocess.call with shell=False — that goes straight through
+    # CreateProcess and each batch stays well under any limit.
+    AR_BATCH_SIZE = 40
+
+    def _batched_ar_action(target, source, env, _ar=ar):
```

Ruff (0.15.11)

[warning] Missing return type annotation for private function `_batched_ar_action`

(ANN202)